### ISSUE TITLE
PPOv2 trainer, the wandb log is unnormal

### ISSUE THREAD

**Issue by kangqiyue** [huggingface/trl#1762]
Hi, I have run my code on ppov2 trainer, and I found the wandb log is unnormal. I have checked the demo track from the doc:

> https://huggingface.co/docs/trl/en/ppov2_trainer
> https://wandb.ai/huggingface/trl/runs/dd2o3g35/workspace?nw=nwusercostahuang

The log is same as mine, when I run the code in my environment. How can we fix it ?

![image](https://github.com/huggingface/trl/assets/64732487/957d26ac-dfe0-461d-a37f-9e365bcea973)

**Comment by kangqiyue** [comment]
update: 
I have temporarily solved it by reinit the wandb run, as follows:

```        if accelerator.is_main_process:
            run = wandb.init(
            # Set the project where this run will be logged
            project="test_ppo",
            run_name=os.path.basename(config.output_dir),
            # Track hyperparameters and run metadata
            config=to_dict(config))
```
for the log

```
                if self.accelerator.is_main_process:
                    self.store_metrics(metrics, train_eval="train")
                    wandb.log(metrics)
```

it worked for now.

![image](https://github.com/huggingface/trl/assets/64732487/d49e0133-c74a-4b22-bdfe-f0b89ab88e02)

**Comment by kangqiyue** [comment]
I think we could have the same solution as: https://github.com/huggingface/trl/pull/1729/files

**Comment by vwxyzjn** [comment]
Try setting wandb x-axis as `train/episodes`.

**Comment by kangqiyue** [comment]
> Try setting wandb x-axis as `train/episodes`.

Thanks~ I tried it, but it didn't work. So I changed my code following 
> https://github.com/huggingface/trl/pull/1729/files

It worked now

**Comment by vwxyzjn** [comment]
Hmm I am still puzzled. What operating system are you using?

**Comment by kangqiyue** [comment]
> Hmm I am still puzzled. What operating system are you using?

I'm using a Linux server (cenos， centos-release-8.2-2.2004.0.1.el8.x86_64)， A100GPU。 I think this problem is because the handler is not properly handled. we called self.log in the hg trainer:
![image](https://github.com/huggingface/trl/assets/64732487/fba50f94-f206-415c-adf2-5175a4f33ad8)
If we call wandb.log, the log is normal. For the pull from https://github.com/huggingface/trl/pull/1729/files, the author added:
> self.state.epoch = episode / self.train_dataset_len  # used by self.log
I guess: the original handler did not change the state for the trainer (but I'm not sure what handler is doing, feel free to check if I'm right)

**Comment by vwxyzjn** [comment]
In my wandb it looks like this:


https://github.com/huggingface/trl/assets/5555347/b411f35b-58a7-4843-a927-0aa13c958577

**Comment by kangqiyue** [comment]
> In my wandb it looks like this:
> 
>  Screen.Recording.2024-06-24.at.10.18.03.AM.mov

ok, I understand. Thanks for your kind explanation!
